plugins/link.md

```diff
@@ -147,19 +154,28 @@ tinymce.init({
 });
 ```
 
-**Example of a custom asynchronous function**
+**Example of a custom asynchronous callback function**
+
+`link_list` can also take a function that is called when the link dialog is opened. TinyMCE passes this function a `success` callback function, which should be passed an array of link items. This allows for asynchronous and dynamic generation of the list of links.
+
+The following example is a trivial example of how link_list can be used with a callback function. `fetchLinkLists` could be replaced with any function that returns an array of link items. For example, it could be used to dynamically generate a list of link items based on the user or current editor content, or to fetch data from a database.
```

Comment:
> For example, it could be used to dynamically generate a list of link items based on the user or current editor content, or to fetch data from a database.

Maybe:
For example, it could be used to generate a list of link items based on:
- the current user.
- the current editor content.
- data retrieved from a database.